pkg/webhook/clusternetwork/validator.go

```diff
@@ -53,6 +57,17 @@ func (c *CnValidator) Delete(_ *admission.Request, oldObj runtime.Object) error
 		return fmt.Errorf(deleteErr, cn.Name, fmt.Errorf("it's not allowed"))
 	}
 
+	nads, err := c.nadCache.List(StorageNetworkNetAttachDefNamespace, labels.Set(map[string]string{
```

Comment:
This may be redundant.

The following `vlanconfig` checks if any vc on this cluster is still existing, then block it. The `storagenetwork` also relys on one `vlanconfig`. Checking `vlanconfig` seems to be enough here.

On the other hand, below function on Harvester needs to check `VlanConfig` is existing when utilize this `vlanconfig` to carry the storagenetwork.

```
func (v *settingValidator) validateStorageNetworkHelper(value string) error {
```

Reply:
@w13915984028 yes makes sense, if storage network is validated for vlan config to be present then validating at cluster network will be redundant.I will remove that check.Thanks.

Regarding storage network validating if vlan config is present, I have another pull request (testing still in progress)
where I am checking if cluster network is ready (cluster network becomes ready only if at least one vlan config attached) before configuring storage networks.
https://github.com/harvester/harvester/pull/6538